data/org.freedesktop.portal.Notification.xml

```diff
@@ -116,8 +166,211 @@
+            The purpose of the action. Mandatory.
+
+            * ``system.custom-alert``:
+
+              This action will be called by the notification server
+              whenever the notification is shown.
+
+              The purpose of this action is to allow apps to use custom methods of
+              notifying the user, for example, to play audio from a special
+              source like a streaming service or a radio station.
+
+            * ``user.reply-with-text``:
+
+              Inline replies for instant messaging apps.
+              The user-provided text will be added to the response vardict as ``reply-text`` (``s``);
+
+            Additional purposes may be defined by notification servers using
+            ``x-vendor`` prefix e.g. ``x-gnome.class.specific``
+
+          * ``target`` (``v``)
+
+            Target parameter to send along when activating the action.
+
+            This target will be added to the response vardict as ``target`` (``v``);
+
+
+        * ``desktop-file-id`` (``s``)
+
+          An optional ``desktop-file-id``. Mendotory for un-sandboxed apps;
+
+          If set, this must be a valid
+          `Desktop File ID <https://specifications.freedesktop.org/desktop-entry-spec/latest/ar01s02.html#desktop-file-id>`_
+          with ``.desktop`` as a suffix.
+
+          For sandboxed apps, ``desktop-file-id``
+          must start with the app ID and can have a prefix after a period.
+          E.g A sandboxed app with id ``org.libreoffice.LibreOffice``
+          is allowed to specify ``org.libreoffice.LibreOffice.Writer.desktop``.
+
+          The desktop file needs to be installed on the system.
+
+        * ``display-hint`` (``as``)
+
+          An array of ways to display the notification. If none are set, or the notification server has its own policy, it is
+          free to decide how and where to display the notification.
+
+          * ``transient``
+
+            The notification is displayed only as a banner and won't be kept
+            by the server in a tray.
+
+            It's a programmer error to specify ``tray`` at the same time.
+
+          * ``tray``
+
+            No banner for the notification will be displayed and
+            the notification is placed in the tray.
+
+            It's a programmer error to specify ``transient`` at the same time.
+
+          * ``persistent``
+
+            Make the notification persistent in the notification tray.
+            The user can’t dismiss it using the usual close button or gesture.
+
+            Apps are only allowed to display persistent notifications
+            as long as they have a window. Once the last window of an app
+            is closed the persistent notification will be removed.
+
+          * ``hide-on-lockscreen``
+
+            Don't show the notification on the lockscreen.
+
+          * ``hide-content-on-lockscreen``
+
+            Only show the notification ``title`` and ``icon`` on the lockscreen.
```

Comment:
Android shows these as well, and e.g. Signal uses it, but in cases where it's actually security-critical you probably don't want to show the name and icon either. Maybe there should be a way to have it just show something like "N new messages in M chats"?

Reply:
allowing alternate title/sub for locked screen should solve this issue.

Reply:
> Android shows these as well, and e.g. Signal uses it, but in cases where it's actually security-critical you probably don't want to show the name and icon either. Maybe there should be a way to have it just show something like "N new messages in M chats"?

I think it makes sense to have two options:
- don't show a notification at all on the look screen
- Hide all of the notification content (including title and icon) but show that there is a notification from a specific app.

> allowing alternate title/sub for locked screen should solve this issue.

I think the right way to give apps control over what's shown on the lock screen is to have a public alternative notification. Like android has https://developer.android.com/reference/androidx/core/app/NotificationCompat.Builder#setPublicVersion(android.app.Notification)

Reply:
For now we only have two hints, one to hide the notification completely and one to hide only the content, but all of it.